src/audio/openal_soft_wrapper/build.rs

```diff
@@ -12,36 +12,48 @@ fn link_search(path: &Path) {
     println!("cargo:rustc-link-search=native={}", path.to_str().unwrap());
 }
 fn link_lib(lib: &str) {
-    println!("cargo:rustc-link-lib=static={}", lib);
+    // Default to static linking
+    if cfg!(feature = "dynamic") {
+        println!("cargo:rustc-link-lib=dylib={}", lib);
+    } else {
+        println!("cargo:rustc-link-lib=static={}", lib);
+    }
 }
+
 fn link_framework(framework: &str) {
     println!("cargo:rustc-link-lib=framework={}", framework);
 }
 
 fn main() {
-    let package_root = Path::new(env!("CARGO_MANIFEST_DIR"));
-    let workspace_root = package_root.join("../../..");
+    if !cfg!(feature = "dynamic") {
+        let package_root = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let workspace_root = package_root.join("../../..");
 
-    let mut build = cmake::Config::new(workspace_root.join("vendor/openal-soft"));
-    build.define("LIBTYPE", "STATIC");
-    let openal_soft_out = build.build();
+        let mut build = cmake::Config::new(workspace_root.join("vendor/openal-soft"));
+        build.define("LIBTYPE", "STATIC");
+        let openal_soft_out = build.build();
 
+        link_search(&openal_soft_out.join("lib"));
+        // some Linux systems
+        link_search(&openal_soft_out.join("lib64"));
+    }
     // Some dependencies of OpenAL Soft.
```

Comment:
This should also be skipped I think, it's probably specific to static linking (I assume the dynamic library links against everything it needs to already)

Reply:
Or wait, am I wrong? I see you link sndio on Linux below…

Reply:
Yeah, you're right, `sndio` only needs to be linked when building statically. Not sure if that is the case with frameworks (dunno much about Mac), but I'll move it in the static block anyway.